sources/packages/web/src/components/common/Root.vue

```diff
@@ -0,0 +1,115 @@
+<template>
+  <div
+    class="v-main body-background"
+    @mouseover="setLastActivityTime"
+    @click="setLastActivityTime"
+    @keyup="setLastActivityTime"
+  >
+    <slot></slot>
+    <ConfirmExtendTime
+      ref="extendTimeModal"
+      :startTimer="startTimer"
+      :clientIdType="clientType"
+    />
+  </div>
+</template>
+
+<script lang="ts">
+import { useRoute } from "vue-router";
+import { onMounted, ref, onUnmounted, computed } from "vue";
+import { ClientIdType, AppRoutes } from "@/types";
+import { useAuth, ModalDialog, useClientLoader } from "@/composables";
+import {
+  MINIMUM_IDLE_TIME_FOR_WARNING_SUPPORTING_USER,
+  MINIMUM_IDLE_TIME_FOR_WARNING_STUDENT,
+  MINIMUM_IDLE_TIME_FOR_WARNING_INSTITUTION,
+  MINIMUM_IDLE_TIME_FOR_WARNING_AEST,
+} from "@/constants/system-constants";
+import ConfirmExtendTime from "@/components/common/modals/ConfirmExtendTime.vue";
+
+export default {
+  components: { ConfirmExtendTime },
+  props: {
+    clientIdType: {
+      type: String,
+      required: true,
+      default: "" as ClientIdType,
+    },
+  },
+  setup(props: any) {
+    const { executeRenewTokenIfExpired } = useAuth();
+    const route = useRoute();
+    const lastActivityLogin = ref(new Date());
+    const interval = ref();
+    const extendTimeModal = ref({} as ModalDialog<boolean>);
+    const startTimer = ref(false);
+    const { getClientType } = useClientLoader();
+
+    const clientType = computed(() => {
+      return getClientType(props.clientIdType);
+    });
+
+    const minimumIdleTime = computed(() => {
+      switch (props.clientIdType) {
+        case ClientIdType.Student:
+          return MINIMUM_IDLE_TIME_FOR_WARNING_STUDENT;
+        case ClientIdType.Institution:
+          return MINIMUM_IDLE_TIME_FOR_WARNING_INSTITUTION;
+        case ClientIdType.SupportingUsers:
+          return MINIMUM_IDLE_TIME_FOR_WARNING_SUPPORTING_USER;
+        case ClientIdType.AEST:
+          return MINIMUM_IDLE_TIME_FOR_WARNING_AEST;
+        default:
+          return 0.02;
+      }
+    });
+
+    const startIdleCheckerTimer = () => {
+      if (!route.path.includes(AppRoutes.Login)) {
+        /* eslint-disable */
```

Comment:
The eslint could be removed by changing the order of the methods.
![image](https://user-images.githubusercontent.com/61259237/139494392-6b0b5fe8-9d97-4f0b-ac34-cee6444db4d2.png)